app/src/main/java/org/oppia/android/app/topic/conceptcard/ConceptCardFactory.kt

```diff
@@ -0,0 +1,67 @@
+package org.oppia.android.app.topic.conceptcard
+
+import android.os.Bundle
+import org.oppia.android.app.model.ProfileId
+import org.oppia.android.util.extensions.putProto
+import javax.inject.Inject
+
+class ConceptCardFactory @Inject constructor(
```

Comment:
Normally we put this inside of the class that it's creating an instance of (in this case, ConceptCardFragment), and just call it 'Factory' (so that it'd be ConceptCardFragment.Factory).

I think we can also avoid the extra callbacks happening here. ConceptCardFragment can register/unregister with the backstack method by injecting it in ConceptCardFragmentPresenter and calling into the corresponding needed methods in the backstack manager at the correct lifecycle times. All the factory needs to do is decide whether to create a new ConceptCardFragment for callers.